src/platform/Linux/bluez/ChipDeviceScanner.cpp

```diff
@@ -112,39 +112,54 @@ void ChipDeviceScanner::Shutdown()
         },
         this);
 
-    mIsInitialized = false;
+    mCurrentState = ChipDeviceScannerState::SCANNER_UNINITIALIZED;
 }
 
 CHIP_ERROR ChipDeviceScanner::StartScan(System::Clock::Timeout timeout)
 {
     assertChipStackLockedByCurrentThread();
-    VerifyOrReturnError(mIsInitialized, CHIP_ERROR_INCORRECT_STATE);
-    VerifyOrReturnError(!mIsScanning, CHIP_ERROR_INCORRECT_STATE);
 
-    mIsScanning = true; // optimistic, to allow all callbacks to check this
-    if (PlatformMgrImpl().GLibMatterContextInvokeSync(MainLoopStartScan, this) != CHIP_NO_ERROR)
+    mScannerLock.lock();
```

Comment:
This code already uses CHIP lock as indicated by `assertChipStackLockedByCurrentThread`? Can we make sure that all other places that refer to the state or other members also use `StackLock` instead of adding another mutex?
And in case you for some reason need another mutex, can you please use `std::lock_guard` instead of manually unlocking it?

Reply:
Yes indeed, there is no need to lock again in `StartScan`, I will delete it